### ISSUE TITLE
Support running the agent with nvidia runtime

### ISSUE THREAD

**Issue by superseb** [rancher/rancher#11467]
When Docker runtime is set to nvidia, state path is different and agent can't be launched.

```
Runtimes: nvidia runc
Default Runtime: nvidia
```

```
/run/docker/runtime-nvidia/moby/9fc0fc5f76d4e0df4db5a4df9b27ecad762ebcfa1e038601999ad5e7becf06d9/state.json```

**Comment by deniseschannon** [comment]
Available in `rancher/agent:v1.2.10-rc4`

**Comment by galal-hussein** [comment]
Rancher server: v1.6.15-rc6
Rancher agent: v1.2.10-rc5

I was able to verify that rancher agent ran successfully and all infra stacks are up and running:
```
root@ip-172-31-60-72:~# ls /run/docker/runtime-nvidia/moby/
1a7eb0edc8a9184fbe9074f04e4b9aa44935b8b1ee390c382fe012e646e5e859  937fead19d827f8ce6c962483091e0788e80c5aa5f29a2837ad50643f87c9f19
3845727ab370ca0be781646eaba9bc3460e6487069fc9b6ef3f915da79b44b44  9e2f0ac4aefafa25b9097ed43ffa53728458e28fd704d6f88c380fbe98db35cf
4d0526860ff7f63aaeca6f9c2fcfd4de283ba32c71b70249aa0e9654d49cbc42  a7e283b9563f69cb1449537d14bffd9c3c0823268357aa3cf72cc9b786936082
5c1a9a483ebd545a4ea692b7c9a6a7835c9ca46e6187bab393ae51de5ebfa3c6  bc2a2c7f07fb38fa8db8f19bfdb0b7edce1bb2b27dc172652579fe252a5bd1a6
60b0eef20006283764ca6b27211ac519197667a56201e5ba03c00f2d2a389167  cdcdfb965a8f686c952cda9c4e85edab370d4333310c7c670c908363494b7d1d
root@ip-172-31-60-72:~# docker ps
CONTAINER ID        IMAGE                             COMMAND                  CREATED             STATUS              PORTS               NAMES
cdcdfb965a8f        rancher/net:v0.13.11              "/rancher-entrypoint…"   2 minutes ago       Up 2 minutes                            r-ipsec-ipsec-router-3-e22361c6
bc2a2c7f07fb        rancher/net:v0.13.11              "/rancher-entrypoint…"   2 minutes ago       Up 2 minutes                            r-ipsec-ipsec-connectivity-check-3-8bea29d7
4d0526860ff7        rancher/dns:v0.17.2               "/rancher-entrypoint…"   2 minutes ago       Up 2 minutes                            r-network-services-metadata-dns-3-8bb6a51e
3845727ab370        rancher/net:holder                "/.r/r /rancher-entr…"   2 minutes ago       Up 2 minutes                            r-ipsec-ipsec-3-20847cd2
60b0eef20006        rancher/healthcheck:v0.3.5        "/.r/r /rancher-entr…"   2 minutes ago       Up 2 minutes                            r-healthcheck-healthcheck-3-49d88e58
5c1a9a483ebd        rancher/metadata:v0.10.2          "/rancher-entrypoint…"   2 minutes ago       Up 2 minutes                            r-network-services-metadata-3-dc34c66d
9e2f0ac4aefa        rancher/network-manager:v0.7.20   "/rancher-entrypoint…"   2 minutes ago       Up 2 minutes                            r-network-services-network-manager-3-2ebd1f21
937fead19d82        rancher/net:v0.13.11              "/rancher-entrypoint…"   2 minutes ago       Up 2 minutes                            r-ipsec-cni-driver-1-95b14cce
a7e283b9563f        rancher/scheduler:v0.8.3          "/.r/r /rancher-entr…"   2 minutes ago       Up 2 minutes                            r-scheduler-scheduler-1-6e0b32d2
1a7eb0edc8a9        rancher/agent:v1.2.10-rc5         "/run.sh run"            3 minutes ago       Up 3 minutes                            rancher-agent
```
